src/engine/source/builder/src/builders/stage/indexerOutput.cpp

```diff
@@ -0,0 +1,86 @@
+#include "indexerOutput.hpp"
+
+#include <memory>
+#include <stdexcept>
+
+#include "builders/utils.hpp"
+
+namespace builder::builders
+{
+
+base::Expression indexerOutputBuilder(const json::Json& definition,
+                                      const std::shared_ptr<const IBuildCtx>& buildCtx,
+                                      const std::shared_ptr<IIndexerConnector>& iConnector)
+{
+    if (!definition.isObject())
+    {
+        throw std::runtime_error(fmt::format(
+            "Stage '{}' expects an object but got '{}'", syntax::asset::INDEXER_OUTPUT_KEY, definition.typeName()));
```

Comment:
Question: Where are these exceptions supposed to be handled? I think we should use some custom exceptions since these errors seem pretty easy to make

Reply:
Hi @sebasfalcone ,This is intentional, all builders in the construction stage throw exceptions if they fail to build correctly, this is captured much higher up, by the policy manager, who is in charge of creating the operations tree.